### DISCUSSION TITLE
how to run plotman in the background?

### DISCUSSION THREAD

**Discussion by medellin9** [ericaltendorf/plotman#404] [General]
Hi im a linux beginner, please be gentle :) I was wondering how to keep plotman running in the background even after i close my ssh terminal, I think it can be done with daemons but I dont understand how to set that up.

Maybe an idea to integrate a background option?

**Comment by Amrac** [comment]
Hello,
You can use the screen system.
To create a screen use:
screen -R YourScreenName
Once in, you can lauch plotman.
Then to leave, you press CTRL+A+D and you go back to your terminal. You can then exit leaving postman working the in screen.

To go back to the screen, use screen -R YourScreenName
To list the available screen, use screen -ls
If you got disconneted while on a screen, the screen will stay "attached". Use screen -D YourScreenName to get it back.

**Reply by medellin9** [reply]
Yes, that works great. Thank you!

Now how can I stop the plotman? The only way i figured out how to do it is with "kill pid"

**Reply by Amrac** [reply]
When you are on the plotman interactive you can press CTRL+C to abort. :)

**Reply by medellin9** [reply]
Ok will take a look, i haven't used interactive yet only cli.

**Reply by altendky** [reply]
FWIW, we are working on better supporting this.  Screen should not be needed, though it is useful at this point.  I hope to have a service that can be run either manually in the background or via a proper service manager and then have interactive usable as a pure monitor that you can close and open independent of the plot launching and archiving activities.  In the mean time, the option laid out above seems to be used by many.

**Reply by medellin9** [reply]
> 
> 
> When you are on the plotman interactive you can press CTRL+C to abort. :)

Yes CTRL + C does indeed end the interactive mode. But when using `plotman plot` in screen function the only way to stop plotting that I know is to actually kill the PID.

**Comment by syadnom** [comment]
I'm a huge screen user (linux guy from... I dare not so how far back).

I prefer the interactive screen myself.  I have a number of plotters so I fire up screen and run plotman interactive on one.  This allows me to fire up a new screen to check htop or dstat etc etc.    I can also split the terminal screen so I can run a 'progress -m' on my subordinate plotters and watch progress on moves and rsyncs that are running.

I tend to migrate around so having screen lets me pop right back in wherever I am.

oh, screen -dR is my go to.  -d disconnect others and -R attach to existing (or create new if no existing)

'q' quits interactive cleanly.  not that it matters.
